template/flake.nix

```
@@ -2,19 +2,6 @@
   description = "virtual environments";
 
   inputs.devshell.url = "github:numtide/devshell";
-  inputs.flake-utils.url = "github:numtide/flake-utils";
 
-  outputs = { self, flake-utils, devshell, nixpkgs }:
-    flake-utils.lib.eachDefaultSystem (system: {
-      devShell =
-        let pkgs = import nixpkgs {
-          inherit system;
-
-          overlays = [ devshell.overlay ];
-        };
-        in
-        pkgs.devshell.mkShell {
-          imports = [ (pkgs.devshell.importTOML ./devshell.toml) ];
-        };
-    });
+  outputs = { devshell, ... }: devshell.lib.fromTOML ./devshell.toml;
```

```suggestion
  outputs = { devshell, nixpkgs, ... }@inputs: {
    devShell = devshell.lib.flakeTOML inputs ./devshell.toml;
  }
```
This would be a bit more verbose but also mix-and-match a bit better?

---

`flakeTOML` returns  `{ devShell.${system} = mkShell...; }`
Thats why current template has no `{ devShell = `

But user could
```nix
{
  outputs = { devshell, nixpkgs, ... }@inputs:
    (devshell.lib.flakeTOML inputs ./devshel.toml) // {
      defaultPackage.x86_64-linux = nixpkgs.legacyPackages.x86_64-linux.hello;
    };
}
```
or

```nix
{
  outputs = { devshell, nixpkgs, ... }@inputs:{
    defaultPackage.x86_64-linux = nixpkgs.legacyPackages.x86_64-linux.hello;
    devShell.x86_64-linux = (devshell.lib.flakeTOML inputs ./devshel.toml).devShell.x86_64-linux;
  };
}
```